content/src/denylist/DenylistServiceDecorator.ts

```diff
@@ -60,9 +60,12 @@ export class DenylistServiceDecorator implements MetaverseContentService {
     }
   }
 
-  /** Is content is denylisted, then we will return that it is not available */
+  /** If content is denylisted, then we will return that it is not available */
   async isContentAvailable(fileHashes: ContentFileHash[]): Promise<Map<string, boolean>> {
     const availability: Map<ContentFileHash, boolean> = await this.service.isContentAvailable(fileHashes)
+    if (await denylistEmpty(this.denylist)) {
```

Comment:
I think that instead of doing this here, the denylist itself should manage this scenario.

This is what I'm thinking: when the denylist is created or the first time it's used, it can detect if it there is something denylisted or not. If there isn't, then it works as a noop/dummy denylist. If there is something, then it works normally. If something is added to the denylist, it will know and it can change from one mode to the other by itself.

By doing that, you can avoid having that responsibility spread all over this decorator. And if we add a new method, we can't forget to add this `if`